### PR TITLE
Use nm-classic instead of nm when host is Darwin

### PR DESCRIPTION
**Pull request by wereHamster**

### CODE REVIEW

File: src/Settings/Builders/Configure.hs [hunk 1]
```diff
@@ -23,4 +25,11 @@ configureBuilderArgs = mconcat
...
                , "--enable-shared=no" -- TODO: add support for yes
                , "--host=" ++ targetPlatform ]
+
+    -- On OS X, use "nm-classic" instead of "nm" due to a bug in the later.
+    -- See https://ghc.haskell.org/trac/ghc/ticket/11744
+    , builder (Configure ".") ?
+        if System.os == "darwin"
+            then arg "--with-nm=$(xcrun --find nm-classic)"
+            else mempty
     ]
```
Comment (snowleopard) [inline src/Settings/Builders/Configure.hs]: There is a more concise way to express this:
```haskell
System.os == "darwin" ? arg "--with-nm=$(xcrun --find nm-classic)"
```

**Comment by snowleopard** [comment]
Looks good, but please see my comment.

**Comment by wereHamster** [comment]
Updated the commit to use the new syntax.

**Comment by snowleopard** [comment]
Hmm, Travis OSX instance failed:
```
xcrun: error: unable to find utility "nm-classic", not a developer tool or in PATH
[...]
Non optional builder 'nm' is not specified in system.config file.
```
Can we make `--with-nm=$(xcrun --find nm-classic)` conditional, that is, if the lookup fails, we shouldn't pass the `--with-nm=` flag to `configure`.

**Comment by snowleopard** [comment]
As discussed offline, another way to fix the Travis OSX failure is to somehow install `nm-classic` there.

**Comment by hvr** [comment]
You shouldn't use `nm-classic` blindly, only when `nm` is known to be broken. Apple will fix `nm` sooner or later, so that `nm-classic` should rather be considered a transitional workaround. Otoh, unfortunately, we can't fallback to `nm-classic` via Autoconf automatically, as that's against the Autoconf philosophy.
The best we can do in line with Autoconf conventions is to detect that `nm` is broken, and ask the user to provide us a different one.

**Comment by snowleopard** [comment]
> The best we can do in line with Autoconf conventions is to detect that nm is broken, and ask the user to provide us a different one.

@hvr This actually sounds good enough to me. Is this check already implemented? Does `configure` advise to user to rerun it with `"--with-nm=$(xcrun --find nm-classic)"` flag?

> You shouldn't use nm-classic blindly, only when nm is known to be broken. Apple will fix nm sooner or later, so that nm-classic should rather be considered a transitional workaround.

Yes, it is just a temporary workaround. It will be dropped as soon as `nm` is fixed.

**Comment by snowleopard** [comment]
@wereHamster Do you have any ideas how to resolve the Travis failure?

**Comment by wereHamster** [comment]
> Does configure advise to user to rerun it with "--with-nm=$(xcrun --find nm-classic)" flag?

No, this happens much later during build time, after `deriveConstants` is built and executed, only then we find out that `nm` is broken. I'll try to move the check to configure.ac.

As for the travis failure, it seems that travis is running a much older version of Mac OS X and/or XCode. This should all go away once we have a proper check in configure.ac.

On platforms which use `nm` (because not all do, some use `objdump` instead, such as AIX. Note to myself: find out why and document):

 1. Check if `nm` produces the correct result.
 2. If $build == darwin:
   1. Try to locate `nm-classic` and retry with it.
 3. If we don't have a working `nm`, fail hard.

**Comment by snowleopard** [comment]
@wereHamster I've finally merged this -- Travis has got `nm-classic` now, so this works fine. Thank you!

**Pull request merged**